src/mach/templates/partials/endpoints/aws_gateway_cloudfront.tf

```diff
@@ -0,0 +1,107 @@
+{% set aws = site.aws %}
+
+provider "aws" {
+  alias   = "mach-cf-us-east-1"
+  region  = "us-east-1"
+
+  {% if aws.deploy_role_name %}
+  assume_role {
+    role_arn = "arn:aws:iam::{{ aws.account_id }}:role/{{ aws.deploy_role_name }}"
+  }
+  {% endif %}
+}
+
+resource "aws_acm_certificate" "cloudfront_api_gateway" {
+  domain_name       = {{ endpoint.url|tf }}
+  validation_method = "DNS"
+
+  provider = aws.mach-cf-us-east-1
+}
+
+resource "aws_route53_record" "cloudfront_api_gateway_acm_validation" {
```

Comment:
The entire ACM certificate + validation + CNAME record don't have to be duplicated.
They both share the same configurations.
Maybe we can rename `aws_api_gateway` to `aws_endpoints`, set up certificate + validation records and API gateway from there

Reply:
done